packages/plugin-essentials/sources/commands/config/set.ts

```diff
@@ -24,8 +48,28 @@ export default class ConfigSetCommand extends BaseCommand {
     if (typeof setting === `undefined`)
       throw new UsageError(`Couldn't find a configuration settings named "${this.name}"`);
 
-    await Configuration.updateConfiguration(configuration.projectCwd, {
-      [this.name]: this.value,
+    const report = await StreamReport.start({
+      configuration,
+      includeFooter: false,
+      stdout: this.context.stdout,
+    }, async report => {
+      const value: unknown = this.json ? JSON.parse((`{ "${this.name}": ${this.value} }`))[this.name] : this.value;
```

Comment:
Hmm why not just this?
```suggestion
      const value: unknown = this.json ? JSON.parse(this.value) : this.value;
```

Reply:
Good question. I don't even have an answer for why I did this... 😅